src/generate.rs

```diff
@@ -0,0 +1,144 @@
+
+    /// Enables more verbose output.
+    #[clap(long, short, action)]
+    pub verbose: bool,
+
+    /// Pass template values through a file
+    /// Values should be in the format `key=value`, one per line
+    #[clap(long, value_parser)]
+    pub template_values_file: Option<String>,
+
+    /// If silent mode is set all variables will be
+    /// extracted from the template_values_file.
+    /// If a value is missing the project generation will fail
+    #[clap(long, short, requires("name"))]
+    pub silent: bool,
+
+    /// Use specific configuration file. Defaults to $CARGO_HOME/cargo-generate or $HOME/.cargo/cargo-generate
+    #[clap(short, long, parse(from_os_str))]
+    pub config: Option<PathBuf>,
+
+    /// Specify the VCS used to initialize the generated template.
+    #[clap(long, default_value = "git")]
+    pub vcs: cargo_generate::Vcs,
+
+    /// Populates a template variable `crate_type` with value `"lib"`
+    #[clap(long, conflicts_with = "bin")]
+    pub lib: bool,
+
+    /// Populates a template variable `crate_type` with value `"bin"`
+    #[clap(long, conflicts_with = "lib")]
+    pub bin: bool,
+
+    /// Use a different ssh identity
+    #[clap(short = 'i', long = "identity", parse(from_os_str))]
+    pub ssh_identity: Option<PathBuf>,
+
+    /// Define a value for use during template expansion
+    #[clap(long, short, number_of_values = 1)]
+    pub define: Vec<String>,
+
+    /// Will enforce a fresh git init on the generated project
+    #[clap(long)]
+    pub force_git_init: bool,
+}
+
+impl Args {
+    pub fn generate(self, config: Config) -> Result<()> {
+        let project_dir = self.cargo_generate(&config.temporary_project_dir)?;
+
+        let delete_file = generate_delete_file(&project_dir)?;
+
+        let mut subprocesses = start_subprocesses(&config, &project_dir);
+
+        let res = start_shell(&config, &project_dir);
+
+        clean_up(&delete_file, &project_dir, &mut subprocesses)?;
+
+        ensure!(res.is_ok(), "problem within the shell process");
+
+        Ok(())
+    }
+
+    fn cargo_generate(self, destination: &Path) -> Result<PathBuf> {
+        cargo_generate::generate(GenerateArgs {
+            template_path: self.template_path,
+            list_favorites: self.list_favorites,
+            name: self.name,
+            force: self.force,
+            verbose: self.verbose,
+            template_values_file: self.template_values_file,
+            silent: self.silent,
+            config: self.config,
+            vcs: self.vcs,
+            lib: self.lib,
+            bin: self.bin,
+            ssh_identity: self.ssh_identity,
+            define: self.define,
+            // Ignore the `--init` flag since we are using a temporary directory.
+            init: false,
+            destination: Some(destination.to_path_buf()),
+            force_git_init: self.force_git_init,
+            // todo: add `allow_commands`.
+            allow_commands: false,
+        })
```

Comment:
```suggestion
        cargo_generate::generate(GenerateArgs {
            // Ignore the `--init` flag since we are using a temporary directory.
            init: false,
            destination: Some(destination.to_path_buf()),
            // todo: add `allow_commands`.
            allow_commands: false,
            ..self
        })
```

Reply:
:rocket:

Reply:
Oh :(
```
expected struct `GenerateArgs`, found struct `generate::Args`
```

Note that `GenerateArgs`is the struct from `cargo_generate` and `generateArgs` is the struct I use to "hide" the fields to the users.

Do you know another solution to remove these fields from the `GenerateArgs` struct directly?

Reply:
Oh sorry I thought it was a builder for some reason.

No there is no other way around afaik